src/api/campaigns/controller.ts

```diff
@@ -53,7 +57,14 @@ export const createCampaign = async (body: any, next: NextFunction) => {
         }
       });
     } else {
-      await sendMail(mailList.emails, newCampaign.subject, Body, newCampaign.senderMail);
+      while (mailList.emails.length > 0) {

```

Comment:
Why await in every iteration? Maybe make an array of promises and them wait for them to resolve together. Improves speed and efficiency. Though you may have to look into the number of emails allowed by ses.